### PR TITLE
Create Pipeline for go-lib-micro repo

### PR DESCRIPTION
**Pull request by tranchitella**
These independent check-commits and check-license pipelines are meant to
be included by any other GitLab Pipeline in the organization.

Note that from Travis to GitLab, the "go build" step has been removed.
This repo does not produce any binary so the build is unnecesary.

Changelog: None

Signed-off-by: Fabio Tranchitella <fabio@tranchitella.eu>

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+include:
+  - local: '.gitlab-ci-check-commits.yml'
+  - local: '.gitlab-ci-check-license.yml'
```
Comment (lluiscampos) [inline .gitlab-ci.yml]: Sorry I didn't explain clear enough. It is only `mendertesting` which has all these extra pipelines which are used from non-golang repos.

This PR should only add one pipeline (this file, `.gitlab-ci.yml`).

Correct the commit message accordingly.

Comment (tranchitella) [reply .gitlab-ci.yml]: Done

**Comment by tranchitella** [comment]
@lluiscampos 

I believe the PR is fine, the pipeline is failing because codecov.io answers 400, I think this is something it should be configured on codecov.io:

```
 https://codecov.io/upload/v4?package=bash-20191203-57b854c&token=secret&branch=pr_50&commit=bdcff13e4e3a44263fcc408a882237305d8c438d&build=372025186&build_url=&name=&tag=&slug=Northern.tech%2FMender%2Fgo-lib-micro&service=gitlab&flags=&pr=&job=372025186
70 HTTP 400
71 Please provide the repository token to upload reports via `-t :repository-token`
75 ERROR: Job failed: exit code 1
```

**Comment by tranchitella** [comment]
@lluiscampos 

with the help of @merlin-northern we fixed the codecov.io integration and now it's green!

**Comment by lluiscampos** [comment]
Everything seems to be correctly set. However, codecov did not send the coverage report to this PR.

**Pull request merged**